### ISSUE TITLE
Ghost Requested Item

### ISSUE THREAD

**Issue by ailtisWSFR** [worldskills/worldskills-infrastructure-list#100]
Hi, 
I've noticed a new bug in the system. When I use the "Requested Item" Tool, I find a Requested Item that I can not find anywhere else ( not appearing in the "Public View" or in the "Skills View" where you can edit directly Requested Items, but appearing in the "Progress" bar of the Skill's overview). 
See video linked below : 
https://www.dropbox.com/s/y5ahmq25jt4b7xd/Ghost%20requested%20item.mp4?dl=0

**Comment by fabian** [comment]
The item is not displayed because it has the same "item_order" in the database as another requested item in that list.

We need to figure out why this can happen, how to prevent it from happening and maybe reconsider the usage of Hibernate `<index column="item_order" />`.

As a quick fix I have manually adjusted "item_order" in the database so the requested item in the video is now visible again at the end of the page on: https://il.worldskills.org/#/event/438/skill/562/category/423

**Comment by fabian** [comment]
Probably happened during mass editing of category.

11 Print
14 Aircraft Maintenance

**Comment by ailtisWSFR** [comment]
Two issues resulting probably of the use of mass editing of categories: 

1- No access on the "Skill view" to categories "Workshop Installation" and "Materials / Consumables", categories involved into the mass editing action. 
![image](https://user-images.githubusercontent.com/39295271/41296292-e9f61af0-6e5c-11e8-9122-f91967a3eafd.png)

2- No access to the "public view":
![image](https://user-images.githubusercontent.com/39295271/41296478-4bd2e1f4-6e5d-11e8-8c20-20f8e40fd3b9.png)

**Comment by fabian** [comment]
Yes, I can confirm that both issues exist because the sequence of `item_order` is not complete.

Most likely a race condition caused by the mass edit tool and the transaction not working properly, ending up with inconsistent data.

Fixing it either involves replacing `<index column="item_order" />` with another sorting method or making sure the transaction is done properly.

@PierreAnctil Do you consider this part of warranty work for the mass editing tool or should we have a look at it?

**Comment by PierreAnctil** [comment]
@fabian since the mass edit tool is calling multiple times the single edit route (as it was done in the catalog) I think the issue could already occur when several users where editing requested items at the same time. So this is not a new bug of the mass edit tool, even though you surely have way more chances to have it happening when mass editing.

**Comment by fabian** [comment]
I spent a few hours debugging this today and I couldn't find a simple solution to make it work.

At least I've written now a migration script to fix the current `item_order` for the skills that are currently broken.

I will test the migration tomorrow on staging and deploy it to production during the day.

**Comment by fabian** [comment]
@ailtisWSFR Okay, the database problem has been fixed now and Skill 11 and 14 can be accessed again.

Please note that the mass editing tool can still cause the same problems again when you change the category as we haven't solved the original problem yet.

@PierreAnctil Can you hide the category field from the mass editing tool for now until we have solved this issue to prevent further data problems?

**Comment by ailtisWSFR** [comment]
@fabian @PierreAnctil 
Any improvement in solving this issue?
